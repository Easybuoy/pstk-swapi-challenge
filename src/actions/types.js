export const LOADING = 'LOADING';
export const ERROR = 'ERROR';
export const SET_MOVIES = 'SET_MOVIES';
export const SET_MOVIE = 'SET_MOVIE';
export const SET_CHARACTERS = 'SET_CHARACTERS';
export const SELECT_MOVIE = 'SELECT_MOVIE';
