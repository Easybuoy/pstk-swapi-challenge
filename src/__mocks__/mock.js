export default {
  getMovieMock: {
    title: 'The Empire Strikes Back',
    episode_id: 5,
    opening_crawl:
      'It is a dark time for the\r\nRebellion. Although the Death\r\nStar has been destroyed,\r\nImperial troops have driven the\r\nRebel forces from their hidden\r\nbase and pursued them across\r\nthe galaxy.\r\n\r\nEvading the dreaded Imperial\r\nStarfleet, a group of freedom\r\nfighters led by Luke Skywalker\r\nhas established a new secret\r\nbase on the remote ice world\r\nof Hoth.\r\n\r\nThe evil lord Darth Vader,\r\nobsessed with finding young\r\nSkywalker, has dispatched\r\nthousands of remote probes into\r\nthe far reaches of space....',
    director: 'Irvin Kershner',
    producer: 'Gary Kurtz, Rick McCallum',
    release_date: '1980-05-17',
    characters: [
      'https://swapi.co/api/people/1/',
      'https://swapi.co/api/people/2/',
      'https://swapi.co/api/people/3/',
      'https://swapi.co/api/people/4/',
      'https://swapi.co/api/people/5/',
      'https://swapi.co/api/people/10/',
      'https://swapi.co/api/people/13/',
      'https://swapi.co/api/people/14/',
      'https://swapi.co/api/people/18/',
      'https://swapi.co/api/people/20/',
      'https://swapi.co/api/people/21/',
      'https://swapi.co/api/people/22/',
      'https://swapi.co/api/people/23/',
      'https://swapi.co/api/people/24/',
      'https://swapi.co/api/people/25/',
      'https://swapi.co/api/people/26/'
    ],
    planets: [
      'https://swapi.co/api/planets/4/',
      'https://swapi.co/api/planets/5/',
      'https://swapi.co/api/planets/6/',
      'https://swapi.co/api/planets/27/'
    ],
    starships: [
      'https://swapi.co/api/starships/15/',
      'https://swapi.co/api/starships/10/',
      'https://swapi.co/api/starships/11/',
      'https://swapi.co/api/starships/12/',
      'https://swapi.co/api/starships/21/',
      'https://swapi.co/api/starships/22/',
      'https://swapi.co/api/starships/23/',
      'https://swapi.co/api/starships/3/',
      'https://swapi.co/api/starships/17/'
    ],
    vehicles: [
      'https://swapi.co/api/vehicles/8/',
      'https://swapi.co/api/vehicles/14/',
      'https://swapi.co/api/vehicles/16/',
      'https://swapi.co/api/vehicles/18/',
      'https://swapi.co/api/vehicles/19/',
      'https://swapi.co/api/vehicles/20/'
    ],
    species: [
      'https://swapi.co/api/species/6/',
      'https://swapi.co/api/species/7/',
      'https://swapi.co/api/species/3/',
      'https://swapi.co/api/species/2/',
      'https://swapi.co/api/species/1/'
    ],
    created: '2014-12-12T11:26:24.656000Z',
    edited: '2017-04-19T10:57:29.544256Z',
    url: 'https://swapi.co/api/films/2/'
  },
  getMoviesMock: [
    {
      title: 'The Empire Strikes Back',
      episode_id: 5,
      opening_crawl:
        'It is a dark time for the\r\nRebellion. Although the Death\r\nStar has been destroyed,\r\nImperial troops have driven the\r\nRebel forces from their hidden\r\nbase and pursued them across\r\nthe galaxy.\r\n\r\nEvading the dreaded Imperial\r\nStarfleet, a group of freedom\r\nfighters led by Luke Skywalker\r\nhas established a new secret\r\nbase on the remote ice world\r\nof Hoth.\r\n\r\nThe evil lord Darth Vader,\r\nobsessed with finding young\r\nSkywalker, has dispatched\r\nthousands of remote probes into\r\nthe far reaches of space....',
      director: 'Irvin Kershner',
      producer: 'Gary Kurtz, Rick McCallum',
      release_date: '1980-05-17',
      characters: [
        'https://swapi.co/api/people/1/',
        'https://swapi.co/api/people/2/',
        'https://swapi.co/api/people/3/',
        'https://swapi.co/api/people/4/',
        'https://swapi.co/api/people/5/',
        'https://swapi.co/api/people/10/',
        'https://swapi.co/api/people/13/',
        'https://swapi.co/api/people/14/',
        'https://swapi.co/api/people/18/',
        'https://swapi.co/api/people/20/',
        'https://swapi.co/api/people/21/',
        'https://swapi.co/api/people/22/',
        'https://swapi.co/api/people/23/',
        'https://swapi.co/api/people/24/',
        'https://swapi.co/api/people/25/',
        'https://swapi.co/api/people/26/'
      ],
      planets: [
        'https://swapi.co/api/planets/4/',
        'https://swapi.co/api/planets/5/',
        'https://swapi.co/api/planets/6/',
        'https://swapi.co/api/planets/27/'
      ],
      starships: [
        'https://swapi.co/api/starships/15/',
        'https://swapi.co/api/starships/10/',
        'https://swapi.co/api/starships/11/',
        'https://swapi.co/api/starships/12/',
        'https://swapi.co/api/starships/21/',
        'https://swapi.co/api/starships/22/',
        'https://swapi.co/api/starships/23/',
        'https://swapi.co/api/starships/3/',
        'https://swapi.co/api/starships/17/'
      ],
      vehicles: [
        'https://swapi.co/api/vehicles/8/',
        'https://swapi.co/api/vehicles/14/',
        'https://swapi.co/api/vehicles/16/',
        'https://swapi.co/api/vehicles/18/',
        'https://swapi.co/api/vehicles/19/',
        'https://swapi.co/api/vehicles/20/'
      ],
      species: [
        'https://swapi.co/api/species/6/',
        'https://swapi.co/api/species/7/',
        'https://swapi.co/api/species/3/',
        'https://swapi.co/api/species/2/',
        'https://swapi.co/api/species/1/'
      ],
      created: '2014-12-12T11:26:24.656000Z',
      edited: '2017-04-19T10:57:29.544256Z',
      url: 'https://swapi.co/api/films/2/'
    }
  ],
  getNetworkErrorMock: { message: 'Network Error' },
  getCharactersMock: [{ name: 'Eze', gender: 'male', height: '300' }],
  getMovieDataMock: [{"title":"A New Hope","movie":{"title":"A New Hope","episode_id":4,"opening_crawl":"galaxy....","director":"George Lucas","producer":"Gary Kurtz, Rick McCallum","release_date":"1977-05-25","characters":["https://swapi.co/api/people/1/","https://swapi.co/api/people/2/","https://swapi.co/api/people/3/","https://swapi.co/api/people/4/","https://swapi.co/api/people/5/","https://swapi.co/api/people/6/","https://swapi.co/api/people/7/","https://swapi.co/api/people/8/","https://swapi.co/api/people/9/","https://swapi.co/api/people/10/","https://swapi.co/api/people/12/","https://swapi.co/api/people/13/","https://swapi.co/api/people/14/","https://swapi.co/api/people/15/","https://swapi.co/api/people/16/","https://swapi.co/api/people/18/","https://swapi.co/api/people/19/","https://swapi.co/api/people/81/"],"planets":["https://swapi.co/api/planets/2/","https://swapi.co/api/planets/3/","https://swapi.co/api/planets/1/"],"starships":["https://swapi.co/api/starships/2/","https://swapi.co/api/starships/3/","https://swapi.co/api/starships/5/","https://swapi.co/api/starships/9/","https://swapi.co/api/starships/10/","https://swapi.co/api/starships/11/","https://swapi.co/api/starships/12/","https://swapi.co/api/starships/13/"],"vehicles":["https://swapi.co/api/vehicles/4/","https://swapi.co/api/vehicles/6/","https://swapi.co/api/vehicles/7/","https://swapi.co/api/vehicles/8/"],"species":["https://swapi.co/api/species/5/","https://swapi.co/api/species/3/","https://swapi.co/api/species/2/","https://swapi.co/api/species/1/","https://swapi.co/api/species/4/"],"created":"2014-12-10T14:23:31.880000Z","edited":"2015-04-11T09:46:52.774897Z","url":"https://swapi.co/api/films/1/"},"characters":[{"name":"Luke Skywalker","height":"172","mass":"77","hair_color":"blond","skin_color":"fair","eye_color":"blue","birth_year":"19BBY","gender":"male","homeworld":"https://swapi.co/api/planets/1/","films":["https://swapi.co/api/films/2/","https://swapi.co/api/films/6/","https://swapi.co/api/films/3/","https://swapi.co/api/films/1/","https://swapi.co/api/films/7/"],"species":["https://swapi.co/api/species/1/"],"vehicles":["https://swapi.co/api/vehicles/14/","https://swapi.co/api/vehicles/30/"],"starships":["https://swapi.co/api/starships/12/","https://swapi.co/api/starships/22/"],"created":"2014-12-09T13:50:51.644000Z","edited":"2014-12-20T21:17:56.891000Z","url":"https://swapi.co/api/people/1/"},{"name":"C-3PO","height":"167","mass":"75","hair_color":"n/a","skin_color":"gold","eye_color":"yellow","birth_year":"112BBY","gender":"n/a","homeworld":"https://swapi.co/api/planets/1/","films":["https://swapi.co/api/films/2/","https://swapi.co/api/films/5/","https://swapi.co/api/films/4/","https://swapi.co/api/films/6/","https://swapi.co/api/films/3/","https://swapi.co/api/films/1/"],"species":["https://swapi.co/api/species/2/"],"vehicles":[],"starships":[],"created":"2014-12-10T15:10:51.357000Z","edited":"2014-12-20T21:17:50.309000Z","url":"https://swapi.co/api/people/2/"},{"name":"R2-D2","height":"96","mass":"32","hair_color":"n/a","skin_color":"white, blue","eye_color":"red","birth_year":"33BBY","gender":"n/a","homeworld":"https://swapi.co/api/planets/8/","films":["https://swapi.co/api/films/2/","https://swapi.co/api/films/5/","https://swapi.co/api/films/4/","https://swapi.co/api/films/6/","https://swapi.co/api/films/3/","https://swapi.co/api/films/1/","https://swapi.co/api/films/7/"],"species":["https://swapi.co/api/species/2/"],"vehicles":[],"starships":[],"created":"2014-12-10T15:11:50.376000Z","edited":"2014-12-20T21:17:50.311000Z","url":"https://swapi.co/api/people/3/"},{"name":"Darth Vader","height":"202","mass":"136","hair_color":"none","skin_color":"white","eye_color":"yellow","birth_year":"41.9BBY","gender":"male","homeworld":"https://swapi.co/api/planets/1/","films":["https://swapi.co/api/films/2/","https://swapi.co/api/films/6/","https://swapi.co/api/films/3/","https://swapi.co/api/films/1/"],"species":["https://swapi.co/api/species/1/"],"vehicles":[],"starships":["https://swapi.co/api/starships/13/"],"created":"2014-12-10T15:18:20.704000Z","edited":"2014-12-20T21:17:50.313000Z","url":"https://swapi.co/api/people/4/"},{"name":"Leia Organa","height":"150","mass":"49","hair_color":"brown","skin_color":"light","eye_color":"brown","birth_year":"19BBY","gender":"female","homeworld":"https://swapi.co/api/planets/2/","films":["https://swapi.co/api/films/2/","https://swapi.co/api/films/6/","https://swapi.co/api/films/3/","https://swapi.co/api/films/1/","https://swapi.co/api/films/7/"],"species":["https://swapi.co/api/species/1/"],"vehicles":["https://swapi.co/api/vehicles/30/"],"starships":[],"created":"2014-12-10T15:20:09.791000Z","edited":"2014-12-20T21:17:50.315000Z","url":"https://swapi.co/api/people/5/"},{"name":"Owen Lars","height":"178","mass":"120","hair_color":"brown, grey","skin_color":"light","eye_color":"blue","birth_year":"52BBY","gender":"male","homeworld":"https://swapi.co/api/planets/1/","films":["https://swapi.co/api/films/5/","https://swapi.co/api/films/6/","https://swapi.co/api/films/1/"],"species":["https://swapi.co/api/species/1/"],"vehicles":[],"starships":[],"created":"2014-12-10T15:52:14.024000Z","edited":"2014-12-20T21:17:50.317000Z","url":"https://swapi.co/api/people/6/"},{"name":"Beru Whitesun lars","height":"165","mass":"75","hair_color":"brown","skin_color":"light","eye_color":"blue","birth_year":"47BBY","gender":"female","homeworld":"https://swapi.co/api/planets/1/","films":["https://swapi.co/api/films/5/","https://swapi.co/api/films/6/","https://swapi.co/api/films/1/"],"species":["https://swapi.co/api/species/1/"],"vehicles":[],"starships":[],"created":"2014-12-10T15:53:41.121000Z","edited":"2014-12-20T21:17:50.319000Z","url":"https://swapi.co/api/people/7/"},{"name":"R5-D4","height":"97","mass":"32","hair_color":"n/a","skin_color":"white, red","eye_color":"red","birth_year":"unknown","gender":"n/a","homeworld":"https://swapi.co/api/planets/1/","films":["https://swapi.co/api/films/1/"],"species":["https://swapi.co/api/species/2/"],"vehicles":[],"starships":[],"created":"2014-12-10T15:57:50.959000Z","edited":"2014-12-20T21:17:50.321000Z","url":"https://swapi.co/api/people/8/"},{"name":"Biggs Darklighter","height":"183","mass":"84","hair_color":"black","skin_color":"light","eye_color":"brown","birth_year":"24BBY","gender":"male","homeworld":"https://swapi.co/api/planets/1/","films":["https://swapi.co/api/films/1/"],"species":["https://swapi.co/api/species/1/"],"vehicles":[],"starships":["https://swapi.co/api/starships/12/"],"created":"2014-12-10T15:59:50.509000Z","edited":"2014-12-20T21:17:50.323000Z","url":"https://swapi.co/api/people/9/"},{"name":"Obi-Wan Kenobi","height":"182","mass":"77","hair_color":"auburn, white","skin_color":"fair","eye_color":"blue-gray","birth_year":"57BBY","gender":"male","homeworld":"https://swapi.co/api/planets/20/","films":["https://swapi.co/api/films/2/","https://swapi.co/api/films/5/","https://swapi.co/api/films/4/","https://swapi.co/api/films/6/","https://swapi.co/api/films/3/","https://swapi.co/api/films/1/"],"species":["https://swapi.co/api/species/1/"],"vehicles":["https://swapi.co/api/vehicles/38/"],"starships":["https://swapi.co/api/starships/48/","https://swapi.co/api/starships/59/","https://swapi.co/api/starships/64/","https://swapi.co/api/starships/65/","https://swapi.co/api/starships/74/"],"created":"2014-12-10T16:16:29.192000Z","edited":"2014-12-20T21:17:50.325000Z","url":"https://swapi.co/api/people/10/"},{"name":"Wilhuff Tarkin","height":"180","mass":"unknown","hair_color":"auburn, grey","skin_color":"fair","eye_color":"blue","birth_year":"64BBY","gender":"male","homeworld":"https://swapi.co/api/planets/21/","films":["https://swapi.co/api/films/6/","https://swapi.co/api/films/1/"],"species":["https://swapi.co/api/species/1/"],"vehicles":[],"starships":[],"created":"2014-12-10T16:26:56.138000Z","edited":"2014-12-20T21:17:50.330000Z","url":"https://swapi.co/api/people/12/"},{"name":"Chewbacca","height":"228","mass":"112","hair_color":"brown","skin_color":"unknown","eye_color":"blue","birth_year":"200BBY","gender":"male","homeworld":"https://swapi.co/api/planets/14/","films":["https://swapi.co/api/films/2/","https://swapi.co/api/films/6/","https://swapi.co/api/films/3/","https://swapi.co/api/films/1/","https://swapi.co/api/films/7/"],"species":["https://swapi.co/api/species/3/"],"vehicles":["https://swapi.co/api/vehicles/19/"],"starships":["https://swapi.co/api/starships/10/","https://swapi.co/api/starships/22/"],"created":"2014-12-10T16:42:45.066000Z","edited":"2014-12-20T21:17:50.332000Z","url":"https://swapi.co/api/people/13/"},{"name":"Han Solo","height":"180","mass":"80","hair_color":"brown","skin_color":"fair","eye_color":"brown","birth_year":"29BBY","gender":"male","homeworld":"https://swapi.co/api/planets/22/","films":["https://swapi.co/api/films/2/","https://swapi.co/api/films/3/","https://swapi.co/api/films/1/","https://swapi.co/api/films/7/"],"species":["https://swapi.co/api/species/1/"],"vehicles":[],"starships":["https://swapi.co/api/starships/10/","https://swapi.co/api/starships/22/"],"created":"2014-12-10T16:49:14.582000Z","edited":"2014-12-20T21:17:50.334000Z","url":"https://swapi.co/api/people/14/"},{"name":"Greedo","height":"173","mass":"74","hair_color":"n/a","skin_color":"green","eye_color":"black","birth_year":"44BBY","gender":"male","homeworld":"https://swapi.co/api/planets/23/","films":["https://swapi.co/api/films/1/"],"species":["https://swapi.co/api/species/4/"],"vehicles":[],"starships":[],"created":"2014-12-10T17:03:30.334000Z","edited":"2014-12-20T21:17:50.336000Z","url":"https://swapi.co/api/people/15/"},{"name":"Jabba Desilijic Tiure","height":"175","mass":"1,358","hair_color":"n/a","skin_color":"green-tan, brown","eye_color":"orange","birth_year":"600BBY","gender":"hermaphrodite","homeworld":"https://swapi.co/api/planets/24/","films":["https://swapi.co/api/films/4/","https://swapi.co/api/films/3/","https://swapi.co/api/films/1/"],"species":["https://swapi.co/api/species/5/"],"vehicles":[],"starships":[],"created":"2014-12-10T17:11:31.638000Z","edited":"2014-12-20T21:17:50.338000Z","url":"https://swapi.co/api/people/16/"},{"name":"Wedge Antilles","height":"170","mass":"77","hair_color":"brown","skin_color":"fair","eye_color":"hazel","birth_year":"21BBY","gender":"male","homeworld":"https://swapi.co/api/planets/22/","films":["https://swapi.co/api/films/2/","https://swapi.co/api/films/3/","https://swapi.co/api/films/1/"],"species":["https://swapi.co/api/species/1/"],"vehicles":["https://swapi.co/api/vehicles/14/"],"starships":["https://swapi.co/api/starships/12/"],"created":"2014-12-12T11:08:06.469000Z","edited":"2014-12-20T21:17:50.341000Z","url":"https://swapi.co/api/people/18/"},{"name":"Jek Tono Porkins","height":"180","mass":"110","hair_color":"brown","skin_color":"fair","eye_color":"blue","birth_year":"unknown","gender":"male","homeworld":"https://swapi.co/api/planets/26/","films":["https://swapi.co/api/films/1/"],"species":["https://swapi.co/api/species/1/"],"vehicles":[],"starships":["https://swapi.co/api/starships/12/"],"created":"2014-12-12T11:16:56.569000Z","edited":"2014-12-20T21:17:50.343000Z","url":"https://swapi.co/api/people/19/"},{"name":"Raymus Antilles","height":"188","mass":"79","hair_color":"brown","skin_color":"light","eye_color":"brown","birth_year":"unknown","gender":"male","homeworld":"https://swapi.co/api/planets/2/","films":["https://swapi.co/api/films/6/","https://swapi.co/api/films/1/"],"species":["https://swapi.co/api/species/1/"],"vehicles":[],"starships":[],"created":"2014-12-20T19:49:35.583000Z","edited":"2014-12-20T21:17:50.493000Z","url":"https://swapi.co/api/people/81/"}],"created_at":1573639860669}]
};
