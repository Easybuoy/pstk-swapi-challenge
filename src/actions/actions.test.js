import configureStore from 'redux-mock-store';
import moxios from 'moxios';
import thunk from 'redux-thunk';

import { getMovies, getMovie, getCharacter, selectMovie } from './index';
import { LOADING, SELECT_MOVIE } from './types';
import mock from '../__mocks__/mock';

describe('Actions', () => {
  const mockStore = configureStore([thunk]);
  const store = mockStore({});

  beforeEach(() => {
    moxios.install();
    store.clearActions();
  });
  afterEach(() => {
    moxios.uninstall();
  });

  it(`dispatches LOADING  when getMovies request is successful`, done => {
    moxios.wait(() => {
      const request = moxios.requests.mostRecent();
      request.respondWith({
        status: 200,
        response: { data: mock.setCurrentUserMock }
      });
    });
    const expectedActions = [
      {
        type: LOADING
      }
    ];

    store.dispatch(getMovies());
    expect(store.getActions()).toEqual(expectedActions);
    expect(store.getActions()).toMatchSnapshot();
    done();
  });

  it(`dispatches LOADING  when getmovie request is successful`, done => {
    moxios.wait(() => {
      const request = moxios.requests.mostRecent();
      request.respondWith({
        status: 200,
        response: { data: mock.setCurrentUserMock }
      });
    });
    const expectedActions = [
      {
        type: LOADING
      }
    ];
    store.dispatch(getMovie('google.com'));
    expect(store.getActions()).toEqual(expectedActions);
    expect(store.getActions()).toMatchSnapshot();
    done();
  });

  it(`dispatches LOADING  when getCharacter request is successful`, done => {
    moxios.wait(() => {
      const request = moxios.requests.mostRecent();
      request.respondWith({
        status: 200,
        response: { data: mock.setCurrentUserMock }
      });
    });
    const expectedActions = [
      {
        type: LOADING
      }
    ];
    store.dispatch(getCharacter([]));
    expect(store.getActions()).toEqual(expectedActions);
    expect(store.getActions()).toMatchSnapshot();
    done();
  });

  it(`dispatches LOADING  when getCharacter request is successful`, done => {
    moxios.wait(() => {
      const request = moxios.requests.mostRecent();
      request.respondWith({
        status: 200,
        response: { data: mock.setCurrentUserMock }
      });
    });
    const expectedActions = [
      {
        type: SELECT_MOVIE,
        payload: []
      }
    ];
    store.dispatch(selectMovie([]));
    expect(store.getActions()).toEqual(expectedActions);
    expect(store.getActions()).toMatchSnapshot();
    done();
  });
});
