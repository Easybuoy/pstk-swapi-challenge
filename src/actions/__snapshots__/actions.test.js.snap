// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions dispatches LOADING  when getCharacter request is successful 1`] = `
Array [
  Object {
    "type": "LOADING",
  },
]
`;

exports[`Actions dispatches LOADING  when getCharacter request is successful 2`] = `
Array [
  Object {
    "payload": Array [],
    "type": "SELECT_MOVIE",
  },
]
`;

exports[`Actions dispatches LOADING  when getMovies request is successful 1`] = `
Array [
  Object {
    "type": "LOADING",
  },
]
`;

exports[`Actions dispatches LOADING  when getmovie request is successful 1`] = `
Array [
  Object {
    "type": "LOADING",
  },
]
`;
